webservices/server-integration/src/main/java/org/jboss/as/webservices/deployers/AspectDeploymentProcessor.java

```diff
@@ -112,4 +124,33 @@ private static boolean isWebServiceDeployment(final DeploymentUnit unit) {
         return unit.getAttachment(WSAttachmentKeys.DEPLOYMENT_KEY) != null;
     }
 
+
+    // Fix for WFLY-12135. We need to let JBossWS know that there's the only listeners available are HTTPS ones.
+    private boolean isHttpsOnly(final DeploymentPhaseContext phaseContext, final DeploymentUnit unit) {
+        CapabilityServiceSupport serviceSupport = unit.getAttachment(Attachments.CAPABILITY_SERVICE_SUPPORT);
+        ServiceName undertowServiceName = serviceSupport.getCapabilityServiceName("org.wildfly.undertow");
+        ServiceController<?> undertowServiceController = phaseContext.getServiceRegistry().getService(undertowServiceName);
+        UndertowService undertowService = (UndertowService) undertowServiceController.getValue();
+
+        Server defaultServer = null;
+
+        for (Server server : undertowService.getServers()) {
+            if (server.getName().equals("default-server")) {
+                defaultServer = server;
+            }
+        }
+
+        if (defaultServer == null) return false;
+
+        List<String> listeners = new ArrayList<>();
+        for (UndertowListener listener : defaultServer.getListeners()) {
+            listeners.add(listener.getProtocol());
+        }
```

Comment:
Copying the protocols out into a list isn't needed to check if there's just 1 listener whose protocol is 'https'.